solana/ed25519/src/lib.rs

```diff
@@ -177,22 +177,22 @@ fn exists_impl(
 /// <https://github.com/solana-labs/solana/blob/master/sdk/src/ed25519_instruction.rs>.
 #[derive(Clone, Copy, bytemuck::Pod, bytemuck::Zeroable)]
 #[repr(C)]
-struct SignatureOffsets {
+pub struct SignatureOffsets {
     /// Offset to ed25519 signature of 64 bytes.
-    signature_offset: u16,
+    pub signature_offset: u16,
     /// Instruction index to find signature.  We support `u16::MAX` only.
-    signature_instruction_index: u16,
+    pub signature_instruction_index: u16,
     /// Offset to public key of 32 bytes.
-    public_key_offset: u16,
+    pub public_key_offset: u16,
```

Comment:
why do we need an offset?

Reply:
That’s how Solana defines the struct.  We don’t really have a say here.  This is copied from solana_sdk.

Reply:
Reference: https://docs.rs/solana-sdk/latest/src/solana_sdk/ed25519_instruction.rs.html#32-83